utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,88 @@
+package describe
+
+import (
+	"testing"
+
+	meshkitkube "github.com/layer5io/meshkit/utils/kubernetes"
+)
+
+/*
+  The MockDescriber is used in the implementation of the Describe function to create a mock output that is returned when the
+  function is called with specific options
+  it takes in a Client object and a DescriberOptions object, and returns a string and an error
+*/
+type MockDescriber struct {
+	DescribeFunc func(*meshkitkube.Client, DescriberOptions) (string, error)
+}
+
+// Describe method of the MockDescriber simply calls DescribeFunc and returns the result.
+func (m *MockDescriber) Describe(client *meshkitkube.Client, options DescriberOptions) (string, error) {
+	return m.DescribeFunc(client, options)
+}
+
+func TestDescribe(t *testing.T) {
+	//meshkitkube.Client  provides the ability to interact with the Kubernetes API server
+
+	//set up mock client client to return expected responses
+	mockClient := meshkitkube.Client{}
+
+	//create test cases
+	testCases := []struct {
+		Name           string
+		Options        DescriberOptions
+		DescribeFunc   func(*meshkitkube.Client, DescriberOptions) (string, error)
+		ExpectedOutput string
+		ExpectedError  error
+	}{
+		{
+			Name: "describe pod",
+			Options: DescriberOptions{
+				Name:      "test-pod",
+				Namespace: "test-namespace",
+				Type:      Pod,
+			},
+			/*
+				DescribeFunc field is a function that takes in a Client
+				object and a DescriberOptions object, and returns a string and an error
+			*/
+			DescribeFunc: func(client *meshkitkube.Client, options DescriberOptions) (string, error) {
```

Comment:
What is it that we are testing over here? You can't define the function that is to be tested inside of the test. This function is to be tested https://github.com/meshery/meshkit/blob/3f37557a2898d7de8a9548893302ca13e25358df/utils/kubernetes/describe/describe.go#L91
Ideally, you need to create a mock http server using httpmock and configure it to mock kube api server's response.

Reply:
Yeah thats true i will remove the DescribeFunc field , to follow best test practice .
 
thanks for highlighting this i  will create mock http servers for k8 api response

Reply:
The <code>DescribeFunc</code> wasnt used for testing it was the <code>tc.Options</code> i used as a parameter for testing. Instead <code>DescribeFunc()</code>  is a testcase field that creates a MockDescriber method for testing. 
The <code>func (m *MockDescriber) Describe(client *meshkitkube.Client, options DescriberOptions) (string, error) {
	return m.DescribeFunc(client, options)
}</code>
 simply returns the result of the <code>DescribeFunc()</code> instance

Reply:
Ok thanks for the suggestion, i would create a mock http server response for the k8 api.

Reply:
I agree with you here but I believe these codes are still useful. If we assume that `MockDescriber` has a function named `Describe` mock the real `Describe`returns it will be useful. Maybe he does not want to mock any of the API server's responses and only wants to test the function itself.  Although, it looks like the code does not test the target function features. We can change a little bit of this code to test the edge situation for the real `Describe` function.

Is that sound good guys?

Reply:
Maybe we can discuss an outline for the mock test. It determines what's your target. After we define the scope, it would have a clear roadmap for mocking the test suite.

Reply:
The approach of using a K8 api  to return any response can slow down the test suite and make it more difficult to maintain.
like @Revolyssup had said i should create a mock response for the api , which is what i will implement .

Reply:
Ok thanks I will look into it .

Reply:
You are welcome. It is useful for all our repos.